src/db/models/article.js

```diff
@@ -1,39 +1,65 @@
 export default (sequelize, DataTypes) => {
-  const Article = sequelize.define('Article', {
-    id: {
-      allowNull: false,
-      autoIncrement: true,
-      primaryKey: true,
-      type: DataTypes.INTEGER
+  const Article = sequelize.define(
+    'Article',
+    {
+      id: {
+        allowNull: false,
+        autoIncrement: true,
+        primaryKey: true,
+        type: DataTypes.INTEGER
+      },
+      slug: {
+        allowNull: false,
+        type: DataTypes.STRING
+      },
+      title: {
+        allowNull: false,
+        unique: false,
+        type: DataTypes.STRING
+      },
+      description: {
+        allowNull: false,
+        type: DataTypes.STRING
+      },
+      body: {
+        allowNull: false,
+        unique: false,
+        type: DataTypes.TEXT
+      },
+      imgUrl: {
+        allowNull: true,
+        unique: false,
+        type: DataTypes.TEXT
+      },
+      userId: {
+        allowNull: false,
+        type: DataTypes.INTEGER
+      },
+      readtime: {
+        type: DataTypes.STRING
+      }
     },
-    slug: {
-      allowNull: false,
-      type: DataTypes.STRING
-    },
-    title: {
-      allowNull: false,
-      unique: false,
-      type: DataTypes.STRING
-    },
-    description: {
-      allowNull: false,
-      type: DataTypes.STRING
-    },
-    body: {
-      allowNull: false,
-      unique: false,
-      type: DataTypes.TEXT
-    },
-    imgUrl: {
-      allowNull: true,
-      unique: false,
-      type: DataTypes.TEXT
-    },
-    userId: {
-      allowNull: false,
-      type: DataTypes.INTEGER
+    {
+      hooks: {
+        beforeCreate(article) {
+          const articleBody = article.body;
+          const articleBodyArray = articleBody.split(' ');
+          const readTimeInMinutes = Math.round(articleBodyArray.length / 250) + 1;
+          const readTimeString = `${readTimeInMinutes} mins`;
+          article.readtime = readTimeString;
+        },
+        beforeBulkUpdate(article) {
```

Comment:
Hi @Johnsonojo, can u please explain why you opted for `beforeBulkUpdate` and not `beforeUpdate` hook to update the readtime of an article

Reply:
I learnt that there is a difference between getting an instance of the article and updating it, and updating by querying the Article model - which is how the `queryForArticle` method is implemented. `beforeUpdate` is an instance update operation and `beforeBulkUpdate` a bulk update operation. And even with the `where` filter  in the `queryForArticle` method, it would return a single item which in this case will be the article with the particular slug. `beforeUpdate` is an instance hook and so it would be triggered on instance updates only. But here the Article model is been queried hence, the `beforeBulkUpdate` hook been implemented.

Reply:
Interesting. I didn't know this before now. Good job pulling this off.

Reply:
Thanks